### ISSUE TITLE
Change of decimals should only apply to future orders in both front- and backend

### ISSUE THREAD

**Issue by jobthomas** [woocommerce/woocommerce#21313]
**Describe the bug**

If someone changes the number of decimals in the store from 2 to 0 (or any other amount), the past orders change as well for both store manager and customer, _but_ this is not entirely correct information. 

For example, if someone paid 65.6 EUR, it would be displayed as 66 EUR, but that's not what the customer has paid (or the store owner received). 

Not only is it wrong information, it also doesn't work with refunds (especially if the new amount is higher). The automated refunds will indicate that this is higher than the original purchase.

**To Reproduce**
1. Create an order that's in the upper half of the decimals (from x.51 to x.99)
2. Change decimals from 2 to 0
3. Check the order from both the store owner and customer perspective; it doesn't reflect the reality anymore
4. Try to process a refund (e.g. with Braintree) - it will indicate that this isn't possible.

**Comment by claudiulodro** [comment]
We'll need to do some further investigation around this. It is likely not feasible since (re)calculations would happen using the current decimal settings (see #19783).

**Comment by jobthomas** [comment]
Yeah, I do understand the need for connecting the two, but ideally, changing it shouldn't apply to past orders in both back and frontend because it reflects the accounting poorly (and as said, would cause issues with refunds)

**Comment by kloon** [comment]
@SimbaAowhoo not sure what this feature has to do with taxes? WooCommerce support taxes, in fact to a very complex degree.

As mentioned in my previous comment, how often will you change your decimal settings? Most, if not all countries work on the 2 decimal system. Also once you set up your store correctly why would you want to change the decimal settings? Curious to know this.

**Comment by SimbaAowhoo** [comment]
@kloon My apologies, I followed from another issue here and I thought it was the same issue. https://github.com/woocommerce/woocommerce/issues/19783
I've been following decimal points issue for years and I was frustrated to see most of the DP related issues got closed almost instantaneously.
I believe DP settings on currency option should not affect the total tax amount.
For example in my country where GST is 7% and we almost never use decimal points for product prices so DP is set to 0.
Suppose the product price is $50 including GST, the GST is $3.27.
But if DP is set to 0, the particular order's GST would become $3.
I believe it's not sensible at all to round the total tax amount.
When I export the order using an export plugin it's still $3.
We had to set DP to 2 and use the below filter to get around this issue.
//Trim zeros on front-end prices
add_filter('woocommerce_price_trim_zeros','__return_true',11); 
But I believe WC would be much more tax friendly if these settings can be done without hacking the functions.php.
Also there's no way to enter tax inclusive shipping cost as of WC 3.2.6, I'm not even sure if that's fixed too.
Thank you.
